core/trino-main/src/main/java/io/trino/execution/scheduler/PipelinedStageExecution.java

```diff
@@ -367,11 +358,10 @@ private synchronized void updateTaskStatus(TaskStatus taskStatus)
                 fail(new TrinoException(GENERIC_INTERNAL_ERROR, "A task is in the ABORTED state but stage is " + stageState));
                 break;
             case FLUSHING:
-                flushingTasks.add(taskStatus.getTaskId());
+                newFlushingOrFinishedTaskObserved = addFlushingTask(taskStatus.getTaskId());
                 break;
             case FINISHED:
-                finishedTasks.add(taskStatus.getTaskId());
-                flushingTasks.remove(taskStatus.getTaskId());
+                newFlushingOrFinishedTaskObserved = addFinishedTask(taskStatus.getTaskId());
```

Comment:
`newFlushingOrFinishedTaskObserved` is unused

Reply:
It was used in the previous version. I wanted check why we observe to products tests failures:
`Test execution is not finished yet, a deadlock or hang is suspected` 
but we still observe that problem.